### DISCUSSION TITLE
Advanced configuration : client certificates

### DISCUSSION THREAD

**Discussion by cpainchaud** [hassio-addons/addon-nginx-proxy-manager#164] [Q&A]
Hi!

 I would like to add the following to my host:

ssl_client_certificate /etc/nginx/client_certs/ca.crt;
sl_verify_client on;

but the problem is I dont know where to drop my files and what would be the path ...

**Comment by cpainchaud** [comment]
![image](https://user-images.githubusercontent.com/6696638/108516791-ef5ef180-72c6-11eb-9571-e073588e6d44.png)

**Reply by frenck** [reply]
Files can be dropped in the `/ssl` folder, which can be addressed as such as well.

**Reply by cpainchaud** [reply]
Thank you, it's what I thought but the following:

ssl_client_certificate /ssl/ChrisCA.crt;
sl_verify_client on;

will cause the Host to remain 'red' status and not working:
![image](https://user-images.githubusercontent.com/6696638/108517923-4d400900-72c8-11eb-8379-192ee8ff688d.png)

**Reply by cpainchaud** [reply]
no error message:

[2/19/2021] [2:52:24 PM] [Nginx    ] › ℹ  info      Reloading Nginx
[2/19/2021] [3:24:48 PM] [SSL      ] › ℹ  info      Renewing SSL certs close to expiry...
[2/19/2021] [3:24:55 PM] [Nginx    ] › ℹ  info      Reloading Nginx
[2/19/2021] [3:24:55 PM] [SSL      ] › ℹ  info      Renew Complete
[2/19/2021] [3:27:06 PM] [Nginx    ] › ℹ  info      Reloading Nginx
[2/19/2021] [3:27:36 PM] [Nginx    ] › ℹ  info      Reloading Nginx
[2/19/2021] [3:27:48 PM] [Nginx    ] › ℹ  info      Reloading Nginx
[2/19/2021] [3:34:14 PM] [Nginx    ] › ℹ  info      Reloading Nginx

**Reply by frenck** [reply]
That, I dunno. You could try asking with the upstream project, you might have a bug on your hands?

**Reply by cpainchaud** [reply]
ok but i dont see where it says in this project Docker file that /ssl should be mounted? . I see no bind/mount , am I right ?

**Reply by cpainchaud** [reply]
I checked with Portainer, it's mounted on /ssl , thank you !

**Reply by cpainchaud** [reply]
It was a terrible typo on my side!!

ssl_client_certificate /ssl/ChrisCA.crt;
ssl_verify_client on;

that will allow SSL Client check!

**Reply by JoseDesfilis** [reply]
I have the same problem. I want to add the directive 
ssl_client_certificate /etc/nginx/whitehorn-ca.crt; 
ssl_verify_client on; 
ssl_verify_depth 1;

But I can’t find where to leave the certificate so that nginx proxy manager can find it.
